### PR TITLE
Add checks to make sure we don't read off the end of our packet.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
There were two issues here:

1) We never checked that the payload length in the packet is valid
given the packet size.  We now add a check for this, and pass the
correct number of available bytes to
MessageAuthenticationCode::Decode.

2) We never checked that the number of bytes we have available to read
is enough to store a payload header.  We now pass the correct buffer
length to PayloadHeader::Decode, to ensure it doesn't read off the end
of the buffer.

 #### Problem
Missing length checks in code that's decoding the various headers on our `PacketBuffer`s.

 #### Summary of Changes
Add the length checks.

 Fixes https://github.com/project-chip/connectedhomeip/issues/2928

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

File: src/transport/RendezvousSession.cpp [hunk 1]
```diff
@@ -351,8 +352,9 @@ CHIP_ERROR RendezvousSession::HandleSecureMessage(PacketBuffer * msgBuf)
 #endif
     plainText = msgBuf->Start();
 
-    // TODO: We need length checks here!  https://github.com/project-chip/connectedhomeip/issues/2928
-    err = mac.Decode(packetHeader, &data[packetHeader.GetPayloadLength()], kMaxTagLen, &taglen);
+    payloadlen = packetHeader.GetPayloadLength();
+    VerifyOrExit(payloadlen < len, err = CHIP_ERROR_INCORRECT_STATE);
```
Comment (pan-apple) [inline src/transport/RendezvousSession.cpp:356]: Error code could be changed.

File: src/transport/SecureSessionMgr.cpp [hunk 2]
```diff
@@ -271,16 +272,19 @@ void SecureSessionMgrBase::HandleDataReceived(const PacketHeader & packetHeader,
 #endif
         plainText = msg->Start();
 
-        // TODO: We need length checks here!  https://github.com/project-chip/connectedhomeip/issues/2928
-        err = mac.Decode(packetHeader, &data[packetHeader.GetPayloadLength()], kMaxTagLen, &taglen);
+        payloadlen = packetHeader.GetPayloadLength();
+        VerifyOrExit(
+            payloadlen <= len,
```
Comment (pan-apple) [inline src/transport/SecureSessionMgr.cpp:277]: any reason why it `<` in the other file?

`VerifyOrExit(payloadlen < len, err = CHIP_ERROR_INCORRECT_STATE);`

**Review by rwalker-apple — APPROVED** [review]

File: src/transport/RendezvousSession.cpp (hunk 1 above)
Comment (bzbarsky-apple) [reply src/transport/RendezvousSession.cpp:356]: Happy to do that; what would you prefer?

File: src/transport/SecureSessionMgr.cpp (hunk 2 above)
Comment (bzbarsky-apple) [reply src/transport/SecureSessionMgr.cpp:277]: Er, yes.  I wrote them both as `<` first, then realized that `<=` might make more sense and failed to switch both.  Will fix that to be `<=` both places.

**Review by mspang — APPROVED** [review]

File: src/transport/RendezvousSession.cpp (hunk 1 above)
Comment (bzbarsky-apple) [reply src/transport/RendezvousSession.cpp]: https://github.com/project-chip/connectedhomeip/pull/3315 will use a better code here.

**Pull request merged**